<a href="https://colab.research.google.com/github/Tranminhtuan48/pythonbasics/blob/main/Ph%E1%BA%A7n_Suy_Di%E1%BB%85n_Th%E1%BB%91ng_K%C3%AA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Tiến hành phân tích suy diễn thông kê cho tập dữ liệu *Diet_R*.**

*Import các thư viện cần thiết

In [ ]:
import pandas as pd
import xlrd
import numpy as np
from scipy import stats
from google.colab import drive
drive.mount('/content/gdrive')

Mounted at /content/gdrive


In [ ]:
dataset = pd.read_csv('/content/gdrive/MyDrive/Diet.csv')
dataset

Person gender  Age  Height  pre.weight  Diet  weight6weeks
0       25          41     171          60     2          60.0
1       26          32     174         103     2         103.0
2        1      0   22     159          58     1          54.2
3        2      0   46     192          60     1          54.0
4        3      0   55     170          64     1          63.3
..     ...    ...  ...     ...         ...   ...           ...
73      74      1   35     183          83     3          80.2
74      75      1   49     177          84     3          79.9
75      76      1   28     164          85     3          79.7
76      77      1   40     167          87     3          77.8
77      78      1   51     175          88     3          81.9

[78 rows x 7 columns]

QUY TRÌNH PHÂN TÍCH ANOVA MỘT CHIỀU

Kiểm tra các giả định:

- Các mẫu độc lập với nhau
- Các mẫu có cùng phương sai
- Các mẫu đều tuân theo phân phối chuẩn

Xây dựng giả thuyết

$$H_{0}: \mu_{0} = \mu_{1} = ... = \mu_{n}$$
$$H_{a}: \text{Có tối thiểu một cặp giá trị trung bình có giá trị khác nhau}$$

Giá trị thống kê F được sử dụng trong việc ra quyết định của phân tích ANOVA một chiều. Công thức tính như sau:

$$F = \frac{MS_{B}}{MS_{W}}$$

với:

- $MS_{B} = \frac{\sum_{k}n_{k}(\overline{\rm x_{k}} - \overline{\rm x})^{2}}{k - 1}$
- $MS_{W} = \frac{\sum_{k}n_{i,k}( x_{i,k} - \overline{\rm x})^{2}}{k - 1}$

Loại bỏ giả thuyết $H_{0}$ khi chỉ số F > chỉ số F-critical (hoặc giá trị p tương ứng < 0.05)

###**Phát Biểu Bài Toán:**
####Tập dữ liệu này chứa thông tin về 78 người sử dụng một trong ba chế độ ăn kiêng. Từ bảng dữ liệu trên hãy cho biết chế độ ăn kiêng nào là phù hợp nhất cho việc giảm cân? Liệu rằng sự khác biệt về giới tính có ảnh hưởng đến quá trình giảm cân hay không?

###* Tiến hành kiểm định theo từng chế độ ăn kiêng khác nhau (có 3 loại)

####Loại 1

In [ ]:
#Diet1
df1=dataset[dataset["Diet"] == 1] 
df1

Person gender  Age  Height  pre.weight  Diet  weight6weeks
2        1      0   22     159          58     1          54.2
3        2      0   46     192          60     1          54.0
4        3      0   55     170          64     1          63.3
5        4      0   33     171          64     1          61.1
6        5      0   50     170          65     1          62.2
7        6      0   50     201          66     1          64.0
8        7      0   37     174          67     1          65.0
9        8      0   28     176          69     1          60.5
10       9      0   28     165          70     1          68.1
11      10      0   45     165          70     1          66.9
12      11      0   60     173          72     1          70.5
13      12      0   48     156          72     1          69.0
14      13      0   41     163          72     1          68.4
15      14      0   37     167          82     1          81.1
45      15      1   39     168          71     1          71.6
46      16      1   31     158          72     1          70.9
47      17      1   40     173          74     1          69.5
48      18      1   50     160          78     1          73.9
49      19      1   43     162          80     1          71.0
50      20      1   25     165          80     1          77.6
51      21      1   52     177          83     1          79.1
52      22      1   42     166          85     1          81.5
53      23      1   39     166          87     1          81.9
54      24      1   40     190          88     1          84.5

###**Phát biểu giả thuyết:** (với mức ý nghĩa α = 0,05)
$$H_{0} : \mu_{pre_w} = \mu_{W_6}$$

$$H_{A} : \text{Có tối thiểu một cặp giá trị trung bình có giá trị khác nhau} $$

Giả thuyết gốc: "Không có sự khác biệt về giá trị trung bình của cân nặng trước và sau 6 tuần ăn kiêng"

Giả thuyết thay thế: "Có sự khác biệt về giá trị trung bình của cân nặng trước và sau 6 tuần ăn kiêng"

**Tiến hành kiểm định các điều kiện *Phân phối chuẩn, phương sai*:**

Phát biểu giả thuyết với mức ý nghĩa *α = 0.05*:
- Null: "Nhóm pre_weight và weight6week được lấy từ phân phối chuẩn và không có sự khác biệt về phương sai giữa các nhóm" 
- Giả thuyết thay thế: "Nhóm pre_weight và weight6week không được lấy từ phân phối chuẩn và có sự khác biệt về phương sai giữa các nhóm"

In [ ]:
#phan phoi chuan pre.weight
p_value = stats.shapiro(df1["pre.weight"])[1]
print(f"P_value của pre_weight: {p_value}")
if p_value < 0.05:
  print(f"Có bằng chứng cho thấy pre_weight không tuân theo phân phối chuẩn")
else:
  print(f"Không có bằng chứng cho thấy nhóm pre_weight không tuân theo phân phối chuẩn")
#phan phoi chuan weight6weeks
p_value = stats.shapiro(df1["weight6weeks"])[1]
print(f"P_value của weight6weeks: {p_value}")
if p_value < 0.05:
  print(f"Có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn")
else:
  print(f"Không có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn")
#phuong sai
stat_value ,p_value = stats.levene(df1["pre.weight"], df1["weight6weeks"])
print(f"P value: {p_value}")
if p_value > 0.05:
    print(f"Có bằng chứng cho thấy sự đồng nhất phương sai giữa các nhóm")
else:
    print(f"Không có bằng chứng cho thấy có sự đồng nhất phương sai giữa các nhóm")

P_value của pre_weight: 0.4964383840560913
Không có bằng chứng cho thấy nhóm pre_weight không tuân theo phân phối chuẩn
P_value của weight6weeks: 0.5883914828300476
Không có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn
P value: 0.9866204065527969
Có bằng chứng cho thấy sự đồng nhất phương sai giữa các nhóm


**Các nhóm đều độc lập, có cùng phương sai và tuân theo phân phối chuẩn nên ta tiến hành kiểm định ANOVA:**

In [ ]:
f_stat, p_value = stats.f_oneway(df1["pre.weight"], df1["weight6weeks"])
print(f"P value: {p_value}")
if p_value < 0.05:
    print(f"Không có bằng chứng để bác bỏ giả thuyết H0")
else:
    print(f"Có bằng chứng để bác bỏ giả thuyết H0")

P value: 0.17972305008882755
Có bằng chứng để bác bỏ giả thuyết H0


Vậy có bằng chứng cho thấy có sự khác biệt về cân nặng trung bình sau 6 tuần ăn kiêng theo chế độ 1

In [ ]:
#tính giá trị trung bình độ lệch cân nặng để so sánh
st1 = df1["pre.weight"].mean() - df1["weight6weeks"].mean()
print(st1)

3.299999999999997


####Loại 2

In [ ]:
#Diet2
df2=dataset[dataset["Diet"] == 2] 
df2

Person gender  Age  Height  pre.weight  Diet  weight6weeks
0       25          41     171          60     2          60.0
1       26          32     174         103     2         103.0
16      27      0   44     174          58     2          60.1
17      28      0   37     172          58     2          56.0
18      29      0   41     165          59     2          57.3
19      30      0   43     171          61     2          56.7
20      31      0   20     169          62     2          55.0
21      32      0   51     174          63     2          62.4
22      33      0   31     163          63     2          60.3
23      34      0   54     173          63     2          59.4
24      35      0   50     166          65     2          62.0
25      36      0   48     163          66     2          64.0
26      37      0   16     165          68     2          63.8
27      38      0   37     167          68     2          63.3
28      39      0   30     161          76     2          72.7
29      40      0   29     169          77     2          77.5
55      41      1   51     191          71     2          66.8
56      42      1   38     199          75     2          72.6
57      43      1   54     196          75     2          69.2
58      44      1   33     190          76     2          72.5
59      45      1   45     160          78     2          72.7
60      46      1   37     194          78     2          76.3
61      47      1   44     163          79     2          73.6
62      48      1   40     171          79     2          72.9
63      49      1   37     198          79     2          71.1
64      50      1   39     180          80     2          81.4
65      51      1   31     182          80     2          75.7

###**Phát biểu giả thuyết:** (với mức ý nghĩa α = 0,05)
$$H_{0} : \mu_{pre_w} = \mu_{W_6}$$

$$H_{A} : \text{Có tối thiểu một cặp giá trị trung bình có giá trị khác nhau} $$

Giả thuyết gốc: "Không có sự khác biệt về giá trị trung bình của cân nặng trước và sau 6 tuần ăn kiêng"

Giả thuyết thay thế: "Có sự khác biệt về giá trị trung bình của cân nặng trước và sau 6 tuần ăn kiêng"

**Tiến hành kiểm định các điều kiện *Phân phối chuẩn, phương sai*:**

Phát biểu giả thuyết với mức ý nghĩa *α = 0.05*:
- Null: "Nhóm pre_weight và weight6week được lấy từ phân phối chuẩn và không có sự khác biệt về phương sai giữa các nhóm" 
- Giả thuyết thay thế: "Nhóm pre_weight và weight6week không được lấy từ phân phối chuẩn và có sự khác biệt về phương sai giữa các nhóm"

In [ ]:
#phan phoi chuan pre.weight
p_value = stats.shapiro(df2["pre.weight"])[1]
print(f"P_value của pre_weight: {p_value}")
if p_value < 0.05:
  print(f"Có bằng chứng cho thấy pre_weight không tuân theo phân phối chuẩn")
else:
  print(f"Không có bằng chứng cho thấy nhóm pre_weight không tuân theo phân phối chuẩn")
#phan phoi chuan weight6weeks
p_value = stats.shapiro(df2["weight6weeks"])[1]
print(f"P_value của weight6weeks: {p_value}")
if p_value < 0.05:
  print(f"Có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn")
else:
  print(f"Không có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn")
#phuong sai
stat_value ,p_value = stats.levene(df2["pre.weight"], df2["weight6weeks"])
print(f"P value: {p_value}")
if p_value > 0.05:
    print(f"Có bằng chứng cho thấy sự đồng nhất phương sai giữa các nhóm")
else:
    print(f"Không có bằng chứng cho thấy có sự đồng nhất phương sai giữa các nhóm")

P_value của pre_weight: 0.005601357202976942
Có bằng chứng cho thấy pre_weight không tuân theo phân phối chuẩn
P_value của weight6weeks: 0.004003484267741442
Có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn
P value: 0.8233570016534023
Có bằng chứng cho thấy sự đồng nhất phương sai giữa các nhóm


**Các nhóm đều độc lập, có cùng phương sai và tuân theo phân phối chuẩn nên ta tiến hành kiểm định ANOVA:**

In [ ]:
f_stat, p_value = stats.f_oneway(df2["pre.weight"], df2["weight6weeks"])
print(f"P value: {p_value}")
if p_value < 0.05:
    print(f"Không có bằng chứng để bác bỏ giả thuyết H0")
else:
    print(f"Có bằng chứng để bác bỏ giả thuyết H0")

P value: 0.27865535599811453
Có bằng chứng để bác bỏ giả thuyết H0


Vậy có bằng chứng cho thấy có sự khác biệt về cân nặng trung bình sau 6 tuần ăn kiêng theo chế độ 2

In [ ]:
#tính giá trị trung bình độ lệch cân nặng để so sánh
st2 = df2["pre.weight"].mean() - df2["weight6weeks"].mean()
print(st2)

3.0259259259259323


####Loại 3

In [ ]:
#Diet3
df3=dataset[dataset["Diet"] == 3] 
df3

Person gender  Age  Height  pre.weight  Diet  weight6weeks
30      52      0   51     165          60     3          53.0
31      53      0   35     169          62     3          56.4
32      54      0   21     159          64     3          60.6
33      55      0   22     169          65     3          58.2
34      56      0   36     160          66     3          58.2
35      57      0   20     169          67     3          61.6
36      58      0   35     163          67     3          60.2
37      59      0   45     155          69     3          61.8
38      60      0   58     141          70     3          63.0
39      61      0   37     170          70     3          62.7
40      62      0   31     170          72     3          71.1
41      63      0   35     171          72     3          64.4
42      64      0   56     171          73     3          68.9
43      65      0   48     153          75     3          68.7
44      66      0   41     157          76     3          71.0
66      67      1   36     155          71     3          68.5
67      68      1   47     179          73     3          72.1
68      69      1   29     166          76     3          72.5
69      70      1   37     173          78     3          77.5
70      71      1   31     177          78     3          75.2
71      72      1   26     179          78     3          69.4
72      73      1   40     179          79     3          74.5
73      74      1   35     183          83     3          80.2
74      75      1   49     177          84     3          79.9
75      76      1   28     164          85     3          79.7
76      77      1   40     167          87     3          77.8
77      78      1   51     175          88     3          81.9

###**Phát biểu giả thuyết:** (với mức ý nghĩa α = 0,05)
$$H_{0} : \mu_{pre_w} = \mu_{W_6}$$

$$H_{A} : \text{Có tối thiểu một cặp giá trị trung bình có giá trị khác nhau} $$

Giả thuyết gốc: "Không có sự khác biệt về giá trị trung bình của cân nặng trước và sau 6 tuần ăn kiêng"

Giả thuyết thay thế: "Có sự khác biệt về giá trị trung bình của cân nặng trước và sau 6 tuần ăn kiêng"

**Tiến hành kiểm định các điều kiện *Phân phối chuẩn, phương sai*:**

Phát biểu giả thuyết với mức ý nghĩa *α = 0.05*:
- Null: "Nhóm pre_weight và weight6week được lấy từ phân phối chuẩn và không có sự khác biệt về phương sai giữa các nhóm" 
- Giả thuyết thay thế: "Nhóm pre_weight và weight6week không được lấy từ phân phối chuẩn và có sự khác biệt về phương sai giữa các nhóm"

In [ ]:
#phan phoi chuan pre.weight
p_value = stats.shapiro(df3["pre.weight"])[1]
print(f"P_value của pre_weight: {p_value}")
if p_value < 0.05:
  print(f"Có bằng chứng cho thấy pre_weight không tuân theo phân phối chuẩn")
else:
  print(f"Không có bằng chứng cho thấy nhóm pre_weight không tuân theo phân phối chuẩn")
#phan phoi chuan weight6weeks
p_value = stats.shapiro(df3["weight6weeks"])[1]
print(f"P_value của weight6weeks: {p_value}")
if p_value < 0.05:
  print(f"Có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn")
else:
  print(f"Không có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn")
#phuong sai
stat_value ,p_value = stats.levene(df3["pre.weight"], df3["weight6weeks"])
print(f"P value: {p_value}")
if p_value > 0.05:
    print(f"Có bằng chứng cho thấy sự đồng nhất phương sai giữa các nhóm")
else:
    print(f"Không có bằng chứng cho thấy có sự đồng nhất phương sai giữa các nhóm")

P_value của pre_weight: 0.7483453750610352
Không có bằng chứng cho thấy nhóm pre_weight không tuân theo phân phối chuẩn
P_value của weight6weeks: 0.35838472843170166
Không có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn
P value: 0.5281844346828702
Có bằng chứng cho thấy sự đồng nhất phương sai giữa các nhóm


**Các nhóm đều độc lập, có cùng phương sai và tuân theo phân phối chuẩn nên ta tiến hành kiểm định ANOVA:**

In [ ]:
f_stat, p_value = stats.f_oneway(df3["pre.weight"], df3["weight6weeks"])
print(f"P value: {p_value}")
if p_value < 0.05:
    print(f"Có bằng chứng để bác bỏ giả thuyết H0")
else:
    print(f"Không có bằng chứng để bác bỏ giả thuyết H0")

P value: 0.020758862909858265
Có bằng chứng để bác bỏ giả thuyết H0


Vậy có bằng chứng cho thấy có sự khác biệt về cân nặng trung bình sau 6 tuần ăn kiêng theo chế độ 3

In [ ]:
#tính giá trị trung bình độ lệch cân nặng để so sánh
st3 = df3["pre.weight"].mean() - df3["weight6weeks"].mean()
print(st3)

5.148148148148138


###Để kiểm tra mối tương quan giữa giới tính và quá trình ăn kiêng, chúng ta cũng áp dụng phương pháp tương tự 

Giả sử

* 0:Nam

* 1:Nữ

In [ ]:
#Nữ
df4=dataset[dataset["gender"] == '1'] 
df4

###**Phát biểu giả thuyết:** (với mức ý nghĩa α = 0,05)
$$H_{0} : \mu_{pre_w} = \mu_{W_6}$$

$$H_{A} : \text{Có tối thiểu một cặp giá trị trung bình có giá trị khác nhau} $$

Giả thuyết gốc: "Không có sự khác biệt về giá trị trung bình của cân nặng trước và sau 6 tuần ăn kiêng"

Giả thuyết thay thế: "Có sự khác biệt về giá trị trung bình của cân nặng trước và sau 6 tuần ăn kiêng"

**Tiến hành kiểm định các điều kiện *Phân phối chuẩn, phương sai*:**

Phát biểu giả thuyết với mức ý nghĩa *α = 0.05*:
- Null: "Nhóm pre_weight và weight6week được lấy từ phân phối chuẩn và không có sự khác biệt về phương sai giữa các nhóm" 
- Giả thuyết thay thế: "Nhóm pre_weight và weight6week không được lấy từ phân phối chuẩn và có sự khác biệt về phương sai giữa các nhóm"

In [ ]:
#phan phoi chuan pre.weight
p_value = stats.shapiro(df4["pre.weight"])[1]
print(f"P_value của pre_weight: {p_value}")
if p_value < 0.05:
  print(f"Có bằng chứng cho thấy pre_weight không tuân theo phân phối chuẩn")
else:
  print(f"Không có bằng chứng cho thấy nhóm pre_weight không tuân theo phân phối chuẩn")
#phan phoi chuan weight6weeks
p_value = stats.shapiro(df4["weight6weeks"])[1]
print(f"P_value của weight6weeks: {p_value}")
if p_value < 0.05:
  print(f"Có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn")
else:
  print(f"Không có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn")
#phuong sai
stat_value ,p_value = stats.levene(df4["pre.weight"], df4["weight6weeks"])
print(f"P value: {p_value}")
if p_value > 0.05:
    print(f"Có bằng chứng cho thấy sự đồng nhất phương sai giữa các nhóm")
else:
    print(f"Không có bằng chứng cho thấy có sự đồng nhất phương sai giữa các nhóm")

P_value của pre_weight: 0.13742199540138245
Không có bằng chứng cho thấy nhóm pre_weight không tuân theo phân phối chuẩn
P_value của weight6weeks: 0.27018821239471436
Không có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn
P value: 0.9058715442431227
Có bằng chứng cho thấy sự đồng nhất phương sai giữa các nhóm


**Các nhóm đều độc lập, có cùng phương sai và tuân theo phân phối chuẩn nên ta tiến hành kiểm định ANOVA:**

In [ ]:
f_stat, p_value = stats.f_oneway(df4["pre.weight"], df4["weight6weeks"])
print(f"P value: {p_value}")
if p_value < 0.05:
    print(f"Có bằng chứng để bác bỏ giả thuyết H0")
else:
    print(f"Không có bằng chứng để bác bỏ giả thuyết H0")

P value: 0.0011404409558938187
Có bằng chứng để bác bỏ giả thuyết H0


In [ ]:
#Nam
df5=dataset[dataset["gender"] == '0'] 
df5

Person gender  Age  Height  pre.weight  Diet  weight6weeks
2        1      0   22     159          58     1          54.2
3        2      0   46     192          60     1          54.0
4        3      0   55     170          64     1          63.3
5        4      0   33     171          64     1          61.1
6        5      0   50     170          65     1          62.2
7        6      0   50     201          66     1          64.0
8        7      0   37     174          67     1          65.0
9        8      0   28     176          69     1          60.5
10       9      0   28     165          70     1          68.1
11      10      0   45     165          70     1          66.9
12      11      0   60     173          72     1          70.5
13      12      0   48     156          72     1          69.0
14      13      0   41     163          72     1          68.4
15      14      0   37     167          82     1          81.1
16      27      0   44     174          58     2          60.1
17      28      0   37     172          58     2          56.0
18      29      0   41     165          59     2          57.3
19      30      0   43     171          61     2          56.7
20      31      0   20     169          62     2          55.0
21      32      0   51     174          63     2          62.4
22      33      0   31     163          63     2          60.3
23      34      0   54     173          63     2          59.4
24      35      0   50     166          65     2          62.0
25      36      0   48     163          66     2          64.0
26      37      0   16     165          68     2          63.8
27      38      0   37     167          68     2          63.3
28      39      0   30     161          76     2          72.7
29      40      0   29     169          77     2          77.5
30      52      0   51     165          60     3          53.0
31      53      0   35     169          62     3          56.4
32      54      0   21     159          64     3          60.6
33      55      0   22     169          65     3          58.2
34      56      0   36     160          66     3          58.2
35      57      0   20     169          67     3          61.6
36      58      0   35     163          67     3          60.2
37      59      0   45     155          69     3          61.8
38      60      0   58     141          70     3          63.0
39      61      0   37     170          70     3          62.7
40      62      0   31     170          72     3          71.1
41      63      0   35     171          72     3          64.4
42      64      0   56     171          73     3          68.9
43      65      0   48     153          75     3          68.7
44      66      0   41     157          76     3          71.0

###**Phát biểu giả thuyết:** (với mức ý nghĩa α = 0,05)
$$H_{0} : \mu_{pre_w} = \mu_{W_6}$$

$$H_{A} : \text{Có tối thiểu một cặp giá trị trung bình có giá trị khác nhau} $$

Giả thuyết gốc: "Không có sự khác biệt về giá trị trung bình của cân nặng trước và sau 6 tuần ăn kiêng"

Giả thuyết thay thế: "Có sự khác biệt về giá trị trung bình của cân nặng trước và sau 6 tuần ăn kiêng"

**Tiến hành kiểm định các điều kiện *Phân phối chuẩn, phương sai*:**

Phát biểu giả thuyết với mức ý nghĩa *α = 0.05*:
- Null: "Nhóm pre_weight và weight6week được lấy từ phân phối chuẩn và không có sự khác biệt về phương sai giữa các nhóm" 
- Giả thuyết thay thế: "Nhóm pre_weight và weight6week không được lấy từ phân phối chuẩn và có sự khác biệt về phương sai giữa các nhóm"

In [ ]:
#phan phoi chuan pre.weight
p_value = stats.shapiro(df5["pre.weight"])[1]
print(f"P_value của pre_weight: {p_value}")
if p_value < 0.05:
  print(f"Có bằng chứng cho thấy pre_weight không tuân theo phân phối chuẩn")
else:
  print(f"Không có bằng chứng cho thấy nhóm pre_weight không tuân theo phân phối chuẩn")
#phan phoi chuan weight6weeks
p_value = stats.shapiro(df5["weight6weeks"])[1]
print(f"P_value của weight6weeks: {p_value}")
if p_value < 0.05:
  print(f"Có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn")
else:
  print(f"Không có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn")
#phuong sai
stat_value ,p_value = stats.levene(df5["pre.weight"], df5["weight6weeks"])
print(f"P value: {p_value}")
if p_value > 0.05:
    print(f"Có bằng chứng cho thấy sự đồng nhất phương sai giữa các nhóm")
else:
    print(f"Không có bằng chứng cho thấy có sự đồng nhất phương sai giữa các nhóm")

P_value của pre_weight: 0.48077937960624695
Không có bằng chứng cho thấy nhóm pre_weight không tuân theo phân phối chuẩn
P_value của weight6weeks: 0.13015852868556976
Không có bằng chứng cho thấy weight6weeks không tuân theo phân phối chuẩn
P value: 0.9168928823683892
Có bằng chứng cho thấy sự đồng nhất phương sai giữa các nhóm


**Các nhóm đều độc lập, có cùng phương sai và tuân theo phân phối chuẩn nên ta tiến hành kiểm định ANOVA:**

In [ ]:
f_stat, p_value = stats.f_oneway(df5["pre.weight"], df5["weight6weeks"])
print(f"P value: {p_value}")
if p_value < 0.05:
    print(f"Có bằng chứng để bác bỏ giả thuyết H0")
else:
    print(f"Không có bằng chứng để bác bỏ giả thuyết H0")

P value: 0.002990831375957425
Có bằng chứng để bác bỏ giả thuyết H0


Vậy có bằng chứng cho thấy giới tính có ảnh hưởng đến chế độ ăn kiêng.

##Kết luận:
- Ta so sánh về mức độ chênh lệch cân năng giữa ba chế độ ăn kiêng khác nhau sau 6 tháng như sau

     5.148148148148138 > 3.299999999999997 > 3.0259259259259323

Vậy điều này ngụ ý rằng chế độ ăn kiêng thứ ba là phù hợp và đem lại hiệu quả tốt nhất. Thêm vào đó sự khác biệt về giới tính cũng là một nhân tố ảnh hưởng đến quá trình ăn kiêng. Do đó tuỳ thuộc vào giới tính và nhu cầu ăn kiêng mà chúng ta có thể chọn chế độ ăn kiêng phù hợp cho bản thân